streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -121,11 +163,13 @@ public void onPartitionsAssigned(Collection<TopicPartition> assignment) {
             try {
                 log.info("stream-thread [{}] New partitions [{}] assigned at the end of consumer rebalance.",
                         StreamThread.this.getName(), assignment);
+                state = StreamThreadStateType.PARTITIONS_ASSIGNED;
                 addStreamTasks(assignment);
                 addStandbyTasks();
                 lastCleanMs = time.milliseconds(); // start the cleaning cycle
                 streamsMetadataState.onChange(partitionAssignor.getPartitionsByHostState(), partitionAssignor.clusterMetadata());
                 initialized.set(true);
```

Comment:
worth it to remove this? you could just use the state variable for the check in `isInitialized`